sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java

```diff
@@ -184,10 +190,14 @@ void startPartitionPump(PartitionOwnership claimedOwnership, Checkpoint checkpoi
             ReceiveOptions receiveOptions = new ReceiveOptions().setOwnerLevel(0L)
                 .setTrackLastEnqueuedEventProperties(trackLastEnqueuedEventProperties);
 
+            Scheduler scheduler = Schedulers.newBoundedElastic(Runtime.getRuntime().availableProcessors(),
+                10000, "partition-pump-" + claimedOwnership.getPartitionId());
```

Comment:
We should consider making this configurable as these numbers may not work for all customers.